great_expectations/data_context/store/tuple_store_backend.py

```diff
@@ -648,14 +648,15 @@ def list_keys(self, prefix: Tuple = ()) -> List[Tuple]:
         return key_list
 
     def get_url_for_key(self, key, protocol=None):
-        location = self._create_client().get_bucket_location(Bucket=self.bucket)[
-            "LocationConstraint"
-        ]
+        location = None
         if self.boto3_options.get("endpoint_url"):
             location = self.boto3_options.get("endpoint_url")
         elif location is None:
             location = "https://s3.amazonaws.com"
         else:
```

Comment:
Based on this, it looks like the `else` condition will never be met, since we set `location = None` on L651. Is there another approach possible here?

Reply:
You are right - I modified the code to the following behaviour:

* if `endpoint_url` is set: use the value for the endpoint
* otherwise:
  1. Retrieve bucket location
  2. build endpoint url using the returned bucket location.